backend/src/connectors/fileScanning/clamAv.ts

```diff
@@ -0,0 +1,88 @@
+import NodeClam from 'clamscan'
+import { Readable } from 'stream'
+
+import { getObjectStream } from '../../clients/s3.js'
+import FileModel, { FileInterfaceDoc, ScanState } from '../../models/File.js'
+import log from '../../services/log.js'
+import config from '../../utils/config.js'
+import { ConfigurationError } from '../../utils/error.js'
+import { BaseFileScanningConnector, FileScanResult } from './Base.js'
+
+let av: NodeClam
+const toolName = 'Clam AV'
+
+export class ClamAvFileScanningConnector extends BaseFileScanningConnector {
+  constructor() {
+    super()
+  }
+
+  async init() {
+    if (!av) {
+      try {
+        av = await new NodeClam().init({ clamdscan: config.avScanning.clamdscan })
+      } catch (error) {
+        throw ConfigurationError('Cannot retrieve Clam AV config', {
+          clamAvConfig: config.avScanning,
+        })
+      }
+    }
+  }
+
+  async scan(file: FileInterfaceDoc) {
+    const avStream = av.passthrough()
```

Comment:
It's a bit of a pain but to be safe, you should probably check that `av` has been initialised at the start of functions like this when using the AV scanner and throw an error recommending that `init()` should be called first if `av` has not been assigned.